File: codegen/codegen-elixir/src/test/resources/ir/aws_json_decode_get_user_request.expected.ex
@doc "Decode AWS JSON request for smithy.beam.test.awsjson11#GetUser."
@spec decode_get_user_request(map()) :: Json11serviceTypes.GetUserInput.t()
def decode_get_user_request(%RuntimeTypes.HttpRequest{body: body}) do
  decoded = decode_json_body(body)
  %Types.GetUserInput{user_name: Map.get(decoded, "userName")}
end

File: codegen/codegen-erlang/src/test/resources/dsl/aws_json_decode_get_user_request.expected.erl
%% @doc Decode AWS JSON request for smithy.beam.test.awsjson11#GetUser.
-spec decode_get_user_request(#http_request{}) -> get_user_input().
decode_get_user_request(#http_request{body = Body}) ->
    Decoded = decode_json_body(Body),
    #get_user_input{
        user_name = maps:get(<<"userName">>, Decoded, undefined)
    }.
